plugins/account_history_plugin/account_history_plugin.cpp

```diff
@@ -0,0 +1,157 @@
+#include <eos/account_history_plugin/account_history_plugin.hpp>
+#include <eos/chain/chain_controller.hpp>
+#include <eos/chain/config.hpp>
+#include <eos/chain/exceptions.hpp>
+#include <eos/chain/transaction.hpp>
+#include <eos/chain/types.hpp>
+
+#include <fc/crypto/sha256.hpp>
+#include <fc/io/json.hpp>
+#include <fc/variant.hpp>
+
+#include <boost/multi_index/hashed_index.hpp>
+#include <boost/multi_index/mem_fun.hpp>
+#include <eos/chain/multi_index_includes.hpp>
+
+namespace eos {
+
+using chain::block_id_type;
+using chain::ProcessedTransaction;
+using chain::signed_block;
+using boost::multi_index_container;
+using chain::transaction_id_type;
+using namespace boost::multi_index;
+
+class transaction_history_object : public chainbase::object<chain::transaction_history_object_type, transaction_history_object> {
+   OBJECT_CTOR(transaction_history_object)
+
+   id_type             id;
+   block_id_type       block_id;
+   transaction_id_type transaction_id;
+};
+
+struct by_id;
+struct by_trx_id;
+using transaction_history_multi_index = chainbase::shared_multi_index_container<
+   transaction_history_object,
+   indexed_by<
+      ordered_unique<tag<by_id>, BOOST_MULTI_INDEX_MEMBER(transaction_history_object, transaction_history_object::id_type, id)>,
+      hashed_unique<tag<by_trx_id>, BOOST_MULTI_INDEX_MEMBER(transaction_history_object, transaction_id_type, transaction_id), std::hash<transaction_id_type>>
+   >
+>;
+
+typedef chainbase::generic_index<transaction_history_multi_index> transaction_history_index;
```

Comment:
Typically we keep the object and index definitions in a header file. Please move to a header for consistency